crypto/ethereum/ethereum.go

```diff
@@ -132,7 +132,7 @@ func (k *SignKeys) AddressString() string {
 
 // SignEthereum signs a message. Message is a normal string (no HexString nor a Hash)
 func (k *SignKeys) SignEthereum(message []byte) ([]byte, error) {
-	if k.Private.D == nil {
+	if k != nil && k.Private.D == nil {
```

Comment:
@mvdan do this check over `k` make sense to you?

Reply:
depends on how the methods are meant to be used. _in general_, it's up to the caller to check that the receiver isn't nil when calling methods on a pointer, so the method itself shouldn't have a nil check. I'd say that applies here too. It's only in cases where the nil value is unavoidable where you want a nil check inside the method, e.g. a `String() string` method which may return `<nil>`.

Reply:
@altergui I think I'd prefer to remove this check. And instead expect the caller is doing the proper check (as all other struct types defined in our code)